packages/web-components/src/components/va-pagination/va-pagination.tsx

```diff
@@ -0,0 +1,232 @@
+import {
+  Component,
+  Event,
+  EventEmitter,
+  Fragment,
+  Host,
+  h,
+  Prop,
+} from '@stencil/core';
+import classnames from 'classnames';
+
+@Component({
+  tag: 'va-pagination',
+  styleUrl: 'va-pagination.css',
+  shadow: true,
+})
+export class VaPagination {
+  /**
+   * Fires when a page is selected
+   */
+  @Event() pageSelect: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. Fires when a
+   * a page is selected if enable-analytics is true.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'component-library-analytics',
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  /**
+   * Aria-label suffix text for buttons
+   */
+  @Prop() ariaLabelSuffix: string = '';
+
+  /**
+   * Whether or not an analytics event will be fired
+   */
+  @Prop() enableAnalytics: boolean = true;
+
+  /**
+   * The maximum number of pages to show at once
+   */
+  @Prop() maxPageListLength: number = 10;
+
+  /**
+   * The current page number
+   */
+  @Prop() page: number;
+
+  /**
+   * The total number of pages
+   */
+  @Prop() pages: number;
+
+  /**
+   * Display last page number when the page count exceeds
+   * `maxPageListLength`
+   */
+  @Prop() showLastPage: boolean;
+
+  private handlePageSelect = (page, eventID) => {
+    this.pageSelect.emit({ page });
+    if (!this.enableAnalytics) return;
+    this.componentLibraryAnalytics.emit({
+      componentName: 'va-pagination',
+      action: 'linkClick',
+      details: {
+        'event': eventID,
+        'paginate-page-number': page,
+      },
+    });
+  };
+
+  private pageNumbers = () => {
+    const {
+      maxPageListLength,
+      page: currentPage,
+      pages: totalPages,
+      showLastPage,
+    } = this;
+
+    // Make space for "... (last page number)" if not in range of the last page.
+    const showEllipsisAndLastPage =
+      showLastPage && currentPage < totalPages - maxPageListLength + 1;
+
+    const limit = showEllipsisAndLastPage
+      ? maxPageListLength - 2
+      : maxPageListLength;
+
+    let end;
+    let start;
+
+    // If there are more pages returned than the limit to show
+    // cap the upper range at limit + the page number.
+    if (totalPages > limit) {
+      start = currentPage;
+      end = limit + currentPage;
+      // treat the last pages specially
+      if (start >= totalPages - limit + 1) {
+        start = totalPages - limit + 1;
+        end = totalPages + 1;
+      }
+    } else {
+      start = 1;
+      end = totalPages + 1;
+    }
+    return Array.from({ length: end - start }, (_, i) => i + start);
+  };
+
+  private handleKeyDown = (e, pageNumber) => {
+    const keyCode = e.key;
+    if (keyCode === 'Enter' || keyCode === ' ') {
```

Comment:
The main motivation for the markup changes is because we have the opportunity to use the correct elements without having to resort to using roles and workarounds such as `tabIndex` to anchors without `href`. This is 100% for assistive technologies. 

Using NVDA, the React component reads out as "Link previous page, link page 1, link page 2, ...link next page". This web component with the markup changes reads out as "Navigation landmark list with 1 item: button previous page, out of list... list with 5 items: button page 1, button page 2, ...out of list, list with 1 items: button next page". I think this adds a lot of value to screen readers.

I wanted to wrap the entire thing in a `ul`, but our pagination component design has a center aligned container that contains the pages. I had to use an `ul` here because using a `div` or `span` inside `ul` fails the axe check.

Reply:
Fantastic, thank you for the explanation.